### ISSUE TITLE
[batch AUC vs streaming AUC]: I want an implement a batch AUC method

### ISSUE THREAD

**Issue by bengxy** [tensorflow/tensorflow#40495]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.4 and 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Tensorflow only support streaming auc. If we want to monitoring the batch auc, we need to reset auc local variables or change the source code of `tensorflow.python.ops.metrics_impl`.

**Will this change the current api? How?**

It only add an optional arg named `batch=False` to current metrics.auc op like:

- tf.keras.metrics.AUC(_, batch=False)

- tf.compat.v1.metrics.auc(_, batch=False)

**Who will benefit with this feature?**
Developers want to monitoring the realtime batch_auc.  E.g:

- online training

- monitoring ordered training/eval data with diff distribution

- compute and log each batch auc

**Any Other info.**

I have implement a batch AUC method for myself and works fine. It changes the code of `tensorflow.python.ops.metrics_impl`. 

I also changed accuracy metric to enable the batch metric function.
